ms2query/spectrum_processing.py

```diff
@@ -0,0 +1,162 @@
+"""Functions to handle spectrum processing steps. Any metadata related cleaning
+and inspection is expected to happen prior to running MS2Query and is not taken
+into account here. Processing here hence refers to inspecting, filtering, adjusting
+the spectrum peaks (m/z and intensities).
+"""
+from typing import Dict, Union
+import numpy as np
+from matchms.typing import SpectrumType
+from matchms.filtering import normalize_intensities, \
+    select_by_mz, select_by_intensity, reduce_to_number_of_peaks, add_losses
+
+
+def set_minimal_processing_defaults(**settings: Dict[str, Union[int, float]]) \
+    -> Dict[str, Union[int, bool]]:
+    """Set default argument values (where no user input is given).
+
+    Args:
+    ----------
+    **settings:
+        Change default settings
+    """
+    defaults = {"mz_from": 10.0,
+                "n_required_below_1000": 5,
+                "intensity_from": 0.001,
+                "max_mz_required": 1000.0,
+                }
+
+    # Set default parameters or replace by **settings input
+    for key in defaults:
+        if key not in settings:
+            settings[key] = defaults[key]
+    return settings
+
+
+def spectrum_processing_minimal(spectrum: SpectrumType,
+                                **settings: Dict[str, Union[int, float]]) \
+    -> Union[SpectrumType, None]:
+    """Minimal necessary spectrum processing that is required by MS2Query.
+    This mostly includes intensity normalization and setting spectrums to None
+    when they do not meet the minimum requirements.
+
+    Args:
+    ----------
+    spectrum:
+        Spectrum to process
+    mz_from
+        Set lower threshold for m/z peak positions. Default is 10.0.
+    n_required_below_1000
+        Number of minimal required peaks with m/z below 1000.0Da for a spectrum
+        to be considered. Spectrums not meeting this criteria will be set to None.
+    intensity_from
+        Set lower threshold for peak intensity. Default is 0.001.
+    max_mz_required
+        Only peaks <= max_mz_required will be counted to check if spectrum
+        contains sufficient peaks to be considered.
+    """
+    settings = set_minimal_processing_defaults(**settings)
+    spectrum = normalize_intensities(spectrum)
+    spectrum = select_by_intensity(spectrum, intensity_from=settings["intensity_from"])
+    spectrum = select_by_mz(spectrum, mz_from=settings["mz_from"], mz_to=np.inf)
+    spectrum = require_peaks_below_mz(spectrum, n_required=settings["n_required_below_1000"],
```

Comment:
Looks good! Just to be sure I understood this correctly, this should be run before loading the spectra into sqlite right? Since none of these processing steps are negative for any other function like ms2ds or cos scores.

Reply:
Yes, that's correct. This should become the minimal processing required for all data going into the sqlite.